### PR TITLE
add version bump script

### PR DESCRIPTION
**Pull request by aidenlx**
Use `npm version` to bump version in `package.json`, `manifest.json` and `version.json`
- `.npmrc` is used to remove the prefix of git tags (`1.0.0` in favor of `v1.0.0`)
- version number in `version.json` is fetched from `minAppVersion` in `manifest.json`

### CODE REVIEW

**Comment by Edo78** [comment]
just a couple of questions:

1. is it possibile to relase such feature specifing that it works only on posix system?
2. I don't have a non posix system so I have no idea about what "unexpected effects" this PR could have. Can you be more specific so that someone can try to solve them?

**Comment by JeppeKlitgaard** [comment]
The command simply fails, I believe. I can't recall if it leaves the respective files in a dirty state, but with Git set up anyway this shouldn't be too harmful.

I think a better option would be to add a JS-based tool to the Obsidian npm package that could do this.

**Comment by aidenlx** [comment]
@JeppeKlitgaard now I've updated to use a simple node script instead of shell script so it should break in non-POSIX environment

for advanced devs they can use solution like [release-it](https://github.com/release-it/release-it) or GitHub Action to do version bump

**Comment by Edo78** [comment]
@aidenlx just wondering why you changed from 4 to 2 spaces (even if I prefer this way I think it's outside the scope of this PR and it made harder to find differences)

File: version-bump.mjs [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import { readFileSync, writeFileSync } from "fs";
+
+const targetVersion = process.env.npm_package_version;
+
+// read minAppVersion from manifest.json and bump version to target version
+let menifest = JSON.parse(readFileSync("manifest.json").toString());
```
Comment (lishid) [inline version-bump.mjs:6]: I think using `readFileSync(path, 'utf8')` makes it a string output directly.

Comment (lishid) [reply version-bump.mjs:6]: Also typo here with `manifest`.

File: version-bump.mjs [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+const { minAppVersion } = menifest;
+menifest.version = targetVersion;
+writeFileSync("manifest.json", JSON.stringify(menifest, null, "\t"));
+
+// update versions.json with target version and minAppVersion from manifest.json
+let versions = JSON.parse(readFileSync("versions.json").toString());
```
Comment (lishid) [inline version-bump.mjs:12]: Same as above.

File: README.md [hunk 3]
```diff
@@ -36,6 +36,9 @@ Quick starting guide for new plugin devs:
 - Upload the files `manifest.json`, `main.js`, `styles.css` as binary attachments. Note: The manifest.json file must be in two places, first the root path of your repository and also in the release.
 - Publish the release.
 
+> You can simplify the version bump process by running `npm version patch`, `npm version minor` or `npm version major` after updating minimum-obsidian-version manually in `manifest.json`.
```
Comment (lishid) [inline README.md:39]: minimum-obsidian-version => minAppVersion

**Review by lishid — CHANGES_REQUESTED** [review]

File: version-bump.mjs (hunk 1 above)
Comment (Edo78) [reply version-bump.mjs:6]: I'm just a passerby but `const` is usually preferable than `let` and you can still modify the properties.
Eg.
```
const obj = {'a':'b'};
obj.a = 'c';
```

File: version-bump.mjs [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+versions[targetVersion] = minAppVersion;
```
Comment (Edo78) [inline version-bump.mjs:13]: Hi, it's me again. Just pointing out that al L8 you wrote `menifest.version` and now `versions[targetVersion]`.
What the preferred style to access to a property?

Comment (lishid) [reply version-bump.mjs:13]: `targetVersion` is a variable 😉

Comment (Edo78) [reply version-bump.mjs:13]: LOL sorry

**Comment by aidenlx** [comment]
Done in [976798](https://github.com/obsidianmd/obsidian-sample-plugin/pull/10/commits/976798a086528d26a8363680d431f48708db7154)

**Comment by lishid** [comment]
Thank you!

**Pull request merged**